test/src/ir_generation/tests/configs_4.sw

```diff
@@ -0,0 +1,29 @@
+script;
+
+const A: u64 = 42;
+const B: u64 = 42;
+
+configurable {
+    X: u64 = 42,
+    Y: u64 = 42,
+}
+
+fn main() -> (u64, u64, u64, u64) {
+    (A, B, X, Y)
+}
+
+// ::check-ir::
+
+// check: script
+
+// ::check-asm::
+
+// regex: DATA=data_\d+
+// regex: REG=\$r\d+
+
+// There should only be 3 data entries here. One shared by `A` and `B` and the
+// other two are for `X` and `Y` respectively
+
+// check: $DATA .word 42
+// check: $DATA .word 42
+// check: $DATA .word 42
```

Comment:
Why don't you give them different values just to be absolutely sure the constants are merged and not the configs.